roles/node_prep/tasks/10_validation.yml

```diff
@@ -184,7 +184,7 @@
         - always
         - validation
   when:
-    - version is ansible.builtin.version('4.12', '>=')
+    - version.split('-')[-1] is ansible.builtin.version('4.12', '>=')
```

Comment:
I'm afraid that this won't work for non-GA builds, e.g.

- Nightly builds, e.g. `4.16.0-0.nightly-2024-05-23-173505`
- RC builds, e.g. `4.16.0-rc.2`
- EC builds, e.g. `4.16.0-ec.6`

Reply:
So it should not be an issue as of the task `Ge` Release.txt File` should fix the rest of issues. This is the only place where the version provided in the playbook is actually used for version checking. Based on the defined format of var `verrion` it should be fine.

Reply:
Without this fix we will have the following issue:

`The error was: Version comparison failed: '<' not supported between instances of 'str' and 'int'\`
 
If the value of the version is set as `latest-4.14` for example

Reply:
In that case, can we move the validation block *after* setting `release_version`? that way we keep the validation always against the `release_version` and not the requested `version`. That is moving:

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/e0977416b11102815693f263819c6c3dc40a63c9/roles/node_prep/tasks/10_validation.yml#L141-L190

To the current line 347:

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/e0977416b11102815693f263819c6c3dc40a63c9/roles/node_prep/tasks/10_validation.yml#L346-L348

Reply:
Yeah we could also do that and always use `release_version` in every validation. It's just that I do not fully know the whole context of if its placed down there for a reason or its just a left over for refactoring.

Reply:
It should be OK to move that validation after, I don't know either why that validation was placed in there, I can only assume because the task before that was also API VIP related, but I see other validations after the set of `relase_version` so I don't expect any issue with that. We will perform tests on it for the case of explicit  versions and for 4.12 and above.

Reply:
I have now made the changes to always use `release_version` to validate.

Reply:
Thanks, I'm performing a few tests with explicit versions. It will take some time. If you could validate the "latest-4.Y" form in the meantime, it would be great.